File: frontend/src/types/product.ts
export type EntityId = string | number

export interface Category {
  id: EntityId
  parentId: EntityId
  name: string
  sort: number
  status: number
}

export interface Product {
  id: EntityId
  categoryId: EntityId
  categoryName: string
  name: string
  subtitle: string
  price: number
  stock: number
  mainImage: string
  detail: string
  status: number
  createdAt?: string
  updatedAt?: string
}

export interface ProductPayload {
  categoryId: EntityId
  name: string
  subtitle: string
  price: number
  stock: number
  mainImage: string
  detail: string
  status: number
}

File: backend/src/main/java/com/example/aicommerce/product/entity/Product.java
package com.example.aicommerce.product.entity;

import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("product")
public class Product {
    @TableId
    private Long id;
    private Long categoryId;
    private String name;
    private String subtitle;
    private BigDecimal price;
    private Integer stock;
    private String mainImage;
    private String detail;
    private Integer status;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

